### PR TITLE
Remove backward compatible changes/upgrades

### PR DESCRIPTION
**Pull request by shubhamgoyal42**
## Why this should be merged
Remove all changes which made it possible to add upgrade in a backward compatible way.
As mentioned [here](https://www.notion.so/hubbleexchange/Breaking-Changes-a0c48331a679496ba8e1a784cc5f860d?pvs=4#58a36dbae18c4c5db504b6c1256e6785)

## How this was tested
On local

### CODE REVIEW

File: plugin/evm/orderbook/order_types.go [hunk 1]
```diff
@@ -16,8 +16,8 @@ type ContractOrder interface {
 	Map() map[string]interface{}
 }
 
-// LimitOrder type is copy of Order struct defined in LimitOrderbook contract
-type LimitOrder struct {
+// OrderCommon is the set of common fields among the order types
+type OrderCommon struct {
```
Comment (shubhamgoyal42) [inline plugin/evm/orderbook/order_types.go:20]: please suggest a better name for this struct

Comment (atvanguard) [reply plugin/evm/orderbook/order_types.go:20]: maybe `BaseOrder` 😛

File: precompile/contracts/juror/eip712.go [hunk 2]
```diff
@@ -9,39 +9,16 @@ import (
...
-		Message:     message,
-	}
-	return EncodeForSigning(typedData)
-}
-
 func GetLimitOrderV2Hash(o *ILimitOrderBookOrderV2) (common.Hash, error) {
-	order := &orderbook.LimitOrderV2{
-		LimitOrder: orderbook.LimitOrder{
+	order := &orderbook.LimitOrder{
+		OrderCommon: orderbook.OrderCommon{
```
Comment (atvanguard) [inline precompile/contracts/juror/eip712.go:21]: For IOC order also we will not use eip712 now.

File: precompile/contracts/juror/eip712.go [hunk 3]
```diff
@@ -54,7 +31,7 @@ func GetLimitOrderV2Hash(o *ILimitOrderBookOrderV2) (common.Hash, error) {
 	return GetLimitOrderV2Hash_2(order)
 }
 
-func GetLimitOrderV2Hash_2(order *orderbook.LimitOrderV2) (common.Hash, error) {
+func GetLimitOrderV2Hash_2(order *orderbook.LimitOrder) (common.Hash, error) {
```
Comment (atvanguard) [inline precompile/contracts/juror/eip712.go:34]: Let's choose a better name.

File: precompile/contracts/juror/logic.go [hunk 4]
```diff
@@ -230,7 +230,7 @@ func validateOrder(bibliophile b.BibliophileClient, orderType OrderType, encoded
 		return validateExecuteIOCOrder(bibliophile, order, side, fillAmount)
 	}
 	if orderType == LimitV2 {
-		order, err := orderbook.DecodeLimitOrderV2(encodedOrder)
+		order, err := orderbook.DecodeLimitOrder(encodedOrder)
```
Comment (atvanguard) [inline precompile/contracts/juror/logic.go:233]: Please remove `LimitV2` enum.

File: precompile/contracts/juror/logic.go [hunk 5]
```diff
@@ -526,23 +526,13 @@ func ValidateCancelLimitOrderV2(bibliophile b.BibliophileClient, inputStruct *Va
 	}
 }
 
-// Sunday, 3 September 2023 10:35:00 UTC
-var V4ActivationDate *big.Int = new(big.Int).SetInt64(1693737300)
-
-func validateCancelLimitOrderV2(bibliophile b.BibliophileClient, order ILimitOrderBookOrderV2, sender common.Address, assertLowMargin bool, blockTimestamp *big.Int) (errorString string, orderHash [32]byte, ammAddress common.Address, unfilledAmount *big.Int) {
+func validateCancelLimitOrderV2(bibliophile b.BibliophileClient, order ILimitOrderBookOrderV2, sender common.Address, assertLowMargin bool) (errorString string, orderHash [32]byte, ammAddress common.Address, unfilledAmount *big.Int) {
```
Comment (atvanguard) [inline precompile/contracts/juror/logic.go:529]: Please remove `V2` from these names.

**Review by atvanguard — COMMENTED** [review]
Please fix tests.

File: plugin/evm/orderbook/memory_database.go [hunk 6]
```diff
@@ -118,7 +118,7 @@ func (order *Order) MarshalJSON() ([]byte, error) {
 	return json.Marshal(&struct {
 		Market                  Market      `json:"market"`
 		PositionType            string      `json:"position_type"`
-		UserAddress             string      `json:"user_address"`
+		Trader                  string      `json:"trader"`
```
[the remark this answers is not in the document]
Comment (shubhamgoyal42) [reply plugin/evm/orderbook/memory_database.go:120]: nope, it's handled in the apis

**Review by atvanguard — APPROVED** [review]

**Pull request merged**